src/compiler/target/golang/mod.rs

```diff
@@ -0,0 +1,268 @@
+      ActionType::StaticNumber(res) => self.action_num(res, lb),
+      ActionType::StaticString(res) => self.action_str(res, lb),
+      ActionType::Variable(res) => self.action_var(res, lb),
+      ActionType::VarRef(res) => lb.code(res),
+      ActionType::While(res) => self.action_while(res, lb),
+      ActionType::If(if_,else_if,else_) => self.action_if(if_,else_if,else_, lb)
+    };
+  }
+  fn if_block(
+    &mut self,
+    lb: &mut impl BuildItems,
+    body: Actions,
+    prefix: &'static str,
+    add_to_prefix: impl FnOnce(&mut Self, &mut Inline),
+  ) {
+    let mut prefix = Inline::from_str(prefix);
+    add_to_prefix(self, &mut prefix);
+    let mut actions = Block::new();
+    for action in body.list {
+      self.action(action, &mut actions, false);
+    }
+    lb.function(prefix, actions);
+  }
+  pub fn action_if(
+    &mut self, 
+    if_: (
+      std::boxed::Box<tokenize::action::Action>,
+      tokenize::actions::Actions,
+    ),
+    else_ifs: std::vec::Vec<(tokenize::action::Action, tokenize::actions::Actions)>,
+    else_: std::option::Option<tokenize::actions::Actions>,
+    lb: &mut impl BuildItems,
+  ) {
+    // if
+    let segment = *if_.0;
+    let body = if_.1;
+    self.if_block(lb, body, "if ", |s, p| {
+      s.action(segment, p, true);
+    });
+
+    // else if
+    for else_if in else_ifs {
+      self.if_block(lb, else_if.1.clone(), "else if ", |s, p| {
+        s.action(else_if.0.clone(), p, true);
+      });
+    }
+
+    // else
+    match else_ {
+      Some(res) => self.if_block(lb, res, "else", |_, _| {}),
+      None => {}
+    }
+  }
+  pub fn action_for(&mut self, action: ActionFor, lb: &mut impl BuildItems) {
+    //   for _, v := range items {
+    //     fmt.Println(v)
+    // }
+    let mut prefix = Inline::from_str(format!("for _, {name} := range ", name = &action.item_name,));
+    self.action(*action.list, &mut prefix, true);
+
+    let mut actions = Block::new();
+    for action in action.actions.list {
+      self.action(action, &mut actions, false);
+    }
+
+    lb.function(prefix, actions);
+  }
+  pub fn action_func_call(
+    &mut self,
+    action: ActionFunctionCall,
+    lb: &mut impl BuildItems,
+  ) {
+    let mut src = Inline::from_str(action.name + "(");
+
+    for (i, arg) in action.arguments.iter().enumerate() {
+      if i != 0 {
+        src.code(",");
+      }
+      self.action(arg.clone(), &mut src, true);
+    }
+
+    src.code(")");
+
+    lb.inline(src);
+  }
+  pub fn action_loop(&mut self, action: Actions, lb: &mut impl BuildItems) {
+    let prefix = Inline::from_str("for true");
+
+    let mut contents = Block::new();
+    for act in action.list {
+      self.action(act, &mut contents, false);
+    }
+
+    lb.function(prefix, contents);
+  }
+  pub fn action_return(&mut self, action: Option<Box<Action>>, lb: &mut impl BuildItems) {
+    let mut src = Inline::from_str("return ");
+
+    self.action(*action.unwrap(), &mut src, true);
```

Comment:
I think it's best here to check if the action exists as we allow to return no data.  
```rust
if let Some(return_action) = action {
  // ...
}
```